packages/errors/lib/upstream-service-error.js

```diff
@@ -29,6 +29,17 @@ class UpstreamServiceError extends HttpError {
 		if (typeof data === 'number') {
 			data = { statusCode: data };
 		}
+
+		// Make sure that we don't modify the original data object
+		// by shallow-cloning it
+		data = { ...data };
+
+		// Default the status code
+		data.statusCode =
+			typeof data.statusCode === 'number'
+				? UpstreamServiceError.normalizeErrorStatusCode(data.statusCode)
```

Comment:
**question:** So if an upstream service returns a 456 error then we set that as the `UpstreamServiceError`'s status code but if no status code is given then it is set to 502 to convey that something has gone wrong upstream?

Reply:
Nope, if an upstream service returns a `456` error then we still send `502`. The conditional here is looking at the `data` object which is supplied by the user, so it's possible to override the `502` but you have to do it explicitly:

```js
throw new UpstreamServiceError({
    // This is `data`
    message: 'Hello',
    statusCode: 456
});
```

I think it's still best to leave the `502` in place in most cases though, and instead pass through a `cause`:

```
throw new UpstreamServiceError({
    message: 'Hello',
    cause: originalFetchError
});
```

because then it'll serialize in a way that's super useful in the logs:

```js
{
    error: {
        name: 'UpstreamServiceError',
        statusCode: 502,
        cause: {
            name: 'FetchError',
            statusCode: 456
        }
    }
}
```

Reply:
Ah, I see - so it's down to advised usage of `UpstreamServiceError` 👍